pkg/alertmanager/state_replication.go

```diff
@@ -134,6 +198,30 @@ func (s *state) Ready() bool {
 	return s.Service.State() == services.Running
 }
 
+// mergeFullStates attempts to merge all full states received from peers during settling.
+func (s *state) mergeFullStates(fs []*clusterpb.FullState) error {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	for _, f := range fs {
+		for _, p := range f.Parts {
+			level.Debug(s.logger).Log("msg", "merging full state", "user", s.userID, "key", p.Key, "bytes", len(p.Data))
+
+			st, ok := s.states[p.Key]
+			if !ok {
+				level.Error(s.logger).Log("msg", "key not found while merging full state", "user", s.userID, "key", p.Key)
+				continue
```

Comment:
From a previous comment. Steve wrote:

> Looking at the code for silences and notifications, it seems the only cause of this error (currently) is failure to decode the message from the wire. Makes sense to fall back to the persisted state, if available.
> I'm actually think if we should do the same for key not found?

If a key has not been found it means that key hasn't been added by `state.AddState()`, which means this alertmanager version doesn't know what to do with that key (eg. the key may have been added by a newer version of the alertmanager and a mix of old and new versions are cohexisting for a while). In this case, I would continue to log but not returning an error.